src/lib/wrapAISDK.ts

```diff
@@ -0,0 +1,229 @@
+const SCORECARD_PROJECT_ID = process.env['SCORECARD_PROJECT_ID'];
+
+/**
+ * Initialize the OpenTelemetry tracer with Scorecard configuration
+ */
+function initializeTracer(config: ScorecardConfig = {}): Tracer {
+  // Return existing tracer if already initialized
+  if (tracer && tracerProvider) {
+    return tracer;
+  }
+
+  const {
+    projectId = SCORECARD_PROJECT_ID,
+    apiKey = SCORECARD_API_KEY,
+    endpoint = 'https://tracing.scorecard.io/otel/v1/traces',
+    serviceName = 'ai-sdk-app',
+    maxExportBatchSize = 1,
+  } = config;
+
+  // Create resource with service information
+  const resource = defaultResource().merge(
+    resourceFromAttributes({
+      [ATTR_SERVICE_NAME]: serviceName,
+      [ATTR_SERVICE_VERSION]: '1.0.0',
+      'scorecard.project_id': projectId,
+    }),
+  );
+
+  // Create span processors
+  const spanProcessors = [];
+
+  try {
+    // Create OTLP exporter for Scorecard
+    const otlpExporter = new OTLPTraceExporter({
+      url: endpoint,
+      headers:
+        apiKey ?
+          {
+            Authorization: `Bearer ${apiKey}`,
+          }
+        : {},
+    });
+
+    // Add batch span processor with OTLP exporter
+    spanProcessors.push(
+      new BatchSpanProcessor(otlpExporter, {
+        maxExportBatchSize,
+      }),
+    );
+  } catch (error) {
+    // Handle initialization errors
+    console.error('Failed to initialize Scorecard telemetry:', error);
+  }
+
+  // Create tracer provider with span processors
+  tracerProvider = new NodeTracerProvider({
+    resource,
+    spanProcessors,
+  });
+
+  // Register the provider
+  tracerProvider.register();
+
+  // Get tracer instance
+  tracer = tracerProvider.getTracer('ai-sdk-wrapper');
+
+  return tracer;
+}
+
+/**
+ * Wraps the AI SDK module to automatically inject telemetry configuration
+ *
+ * @param aiSDKModule - The AI SDK module (e.g., import('ai'))
+ * @param config - Optional Scorecard configuration
+ * @returns Proxied AI SDK module with automatic telemetry injection
+ *
+ * @example
+ * ```typescript
+ * import ai from 'ai';
+ * import { wrapAISDK } from 'scorecard-ai/lib/wrapAISDK';
+ *
+ * const aiSDK = await wrapAISDK(ai);
+ *
+ * // Now all AI SDK calls will automatically send traces to Scorecard
+ * const { text } = await aiSDK.generateText({
+ *   model: openai('gpt-4o-mini'),
+ *   prompt: 'Hello!',
+ * });
+ * ```
+ */
+export async function wrapAISDK<T extends Record<string, any>>(
+  aiSDKModule: T,
+  config: ScorecardConfig = {},
+): Promise<T> {
+  const apiKey = config.apiKey || SCORECARD_API_KEY;
+  if (!apiKey) {
+    throw new Error('SCORECARD_API_KEY environment variable is not set');
+  }
+
+  const baseUrl = new URL(config.apiBaseUrl || 'https://api2.scorecard.io');
+  // Create metrics and monitor if needed
+  if (config.metrics && config.metrics.length > 0 && config.projectId) {
+    fetch(`${baseUrl.origin}/api/v2/projects/${config.projectId}/monitoring/create-if-needed`, {
```

Comment:
make a Scorecard client, then:

```suggestion 
const client = new Scorecard({apiKey, baseUrl});
 await client.post(`/projects/${config.projectId}/monitoring/create-if-needed`, {
        body: {
            description: config.serviceName,
            metrics: config.metrics,
            filter: {
              serviceName: config.serviceName,
            },
        }
        query: {}
    });
```

Reply:
Also, wrap this part in a try-catch block and if the call fails, just keep going (maybe log an error?)

If it's 4xx, maybe throw a blocking error?
If it's 5xx, maybe silently continue?